test_runner/regress/test_metric_collection.py

```diff
@@ -136,3 +150,57 @@ def get_num_remote_ops(file_kind: str, op_kind: str) -> int:
     httpserver.check()
     global num_metrics_received
     assert num_metrics_received > 0, "no metrics were received"
+
+
+# FIXME
+# test to illustrate weird behavior of calculate_synthetic_size with multiple branches
+#
+def test_metric_collection_multiple_branches(
+    httpserver: HTTPServer, neon_env_builder: NeonEnvBuilder, httpserver_listen_address
+):
+
+    (host, port) = httpserver_listen_address
+    metric_collection_endpoint = f"http://{host}:{port}/billing/api/v1/usage_events"
+
+    # configure pageserver to send metrics to the mock http server
+    # setup gc to observe synthetic_storage_size changes
+    gc_horizon = 0x30000
+    neon_env_builder.pageserver_config_override = f"""
+    metric_collection_endpoint="{metric_collection_endpoint}"
+    metric_collection_interval="10s"
+    tenant_config={{gc_period='0s', pitr_interval='0sec', gc_horizon={gc_horizon}}}
+    """
+
+    log.info(f"test_metric_collection endpoint is {metric_collection_endpoint}")
+
+    # mock http server that returns OK for the metrics
+    httpserver.expect_request("/billing/api/v1/usage_events", method="POST").respond_with_handler(
+        metrics_debug
+    )
+
+    # spin up neon, after http server is ready
+    env = neon_env_builder.init_start()
+    env.neon_cli.create_branch("test_metric_collection0")
+    pg = env.postgres.create_start("test_metric_collection0")
+
+    # # request tenant size manually
+    # tenant_id = env.initial_tenant
+    # http_client = env.pageserver.http_client()
+    # size = http_client.tenant_size(tenant_id)
+    # log.info(f"tenant_size is {size}")
+
+    # create 10 branches and 10 tables in each branch
+    for i in range(1, 10):
+        env.neon_cli.create_branch(f"test_metric_collection{i}", f"test_metric_collection{i-1}")
+        pg = env.postgres.create_start(f"test_metric_collection{i}")
+
+        with pg.cursor() as cur:
+            cur.execute(f"CREATE TABLE t{i} as SELECT g FROM generate_series(1, 100000) g")
+
+        log.info(f"created table t{i}")
+        sleep(5)
+
+    # check that all requests are served
+    httpserver.check()
+    global num_metrics_received
+    assert num_metrics_received > 0, "no metrics were received"
```

Comment:
Can we make an assertion here and in other `test_metrics_collection.py` tests on real numbers produced by the metrics collectors? So we could be sure the calculations are close to reality.

Reply:
I will do in a follow-up PR